File: 2022/05/18/is_subsequence.py
"""
https://leetcode.com/problems/is-subsequence/
Author: Andrew Jarombek
Date: 5/18/2022
"""


def is_subsequence(s: str, t: str) -> bool:
    i = 0
    j = 0

    while i < len(s) and j < len(t):
        if s[i] == t[j]:
            i += 1

        j += 1

    return i == len(s)

File: 2023/06/15/go/is_subsequence.go
/**
 * https://leetcode.com/problems/is-subsequence/
 * Author: Andrew Jarombek
 * Date: 6/15/2023
 */

package _go

func isSubsequence(s string, t string) bool {
	i := 0
	j := 0

	for i < len(s) && j < len(t) {
		if s[i] == t[j] {
			i++
		}

		j++
	}

	return i == len(s)
}
